### PR TITLE
Add i18n support and  translation for zh-CN

### PR DESCRIPTION
**Pull request by Sevichecc**
Relative issue: #223

### CODE REVIEW

File: frontend/public/locales/zh-CN/common.json [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    "club_select_label": "俱乐部",
+    "now_button": "现在",
+    "logout_title": "登出",
+    "user_title": "用户",
+    "more_title": "更多",
+    "website_title": "网站",
+    "github_title": "Github",
+    "api_docs_title": "API 文档",
+    "elo_input_label": "最大 ELO 差值",
+    "only_recommended_input_group_label": "只显示比赛次数较少的队伍",
```
Comment (evroon) [inline frontend/public/locales/zh-CN/common.json:180]: The indentation here is slightly off, could you remove the two spaces at the start of each line?

**Review by evroon — COMMENTED** [review]
Cool! Thanks a lot for this, it's very useful

I noticed the build fails:

image

Do you know what might be the problem? Otherwise, I can also look into that in a few days

**Comment by Sevichecc** [comment]
> Cool! Thanks a lot for this, it's very useful
> 
> I noticed the build fails:
> 
> image
> 
> Do you know what might be the problem? Otherwise, I can also look into that in a few days

It might be something wrong with the position of the translation hook, I will try to fix this a bit later.

**Comment by evroon** [comment]
It might also be that Next is trying to build the site with server-side rendering, you could try adding these lines from the docs:

https://react.i18next.com/latest/ssr#setting-the-i18next-instance-based-on-req

**Comment by Sevichecc** [comment]
> It might also be that Next is trying to build the site with server-side rendering, you could try adding these lines from the docs:
> 
> https://react.i18next.com/latest/ssr#setting-the-i18next-instance-based-on-req

The error is quite straightforward, which I encountered during development but didn't give it enough attention. The error was caused by the incorrect placement of the useTranslation hook; it should not be used outside of a functional component.

Additionally, some pages did not render the translation correctly, so I've implemented some fixes for that.

File: frontend/public/locales/zh-CN/common.json (hunk 1 above)
Comment (Sevichecc) [reply frontend/public/locales/zh-CN/common.json]: Fixed

File: frontend/public/locales/en/common.json [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+  "clubs_title": "Clubs",
+  "empty_name_validation": "Name cannot be empty",
+  "empty_email_validation": "Invalid email",
+  "empty_password_validation": "Password cannot be empty",
+  "too_short_name_validation": "Name is too short",
+  "too_short_dashboard_link_validation": "Dashboard link is short",
+  "too_short_password_validation": "Password is too short",
+  "invalid_email_validation": "Invalid email",
+  "invalid_password_validation": "Invalid password",
+  "create_account_title": "Create a new Count",
```
Comment (evroon) [inline frontend/public/locales/en/common.json:14]:
```suggestion
  "create_account_title": "Create a new account",
```

File: frontend/public/locales/zh-CN/common.json [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+  "too_short_password_validation": "密码太短",
+  "invalid_email_validation": "无效的电子邮件",
+  "invalid_password_validation": "无效的密码",
+  "create_account_title": "创建一个新的计数",
+  "create_account_alert_title": "不可用",
+  "create_account_alert_description": "由于括号仍处于测试阶段，因此暂时禁用了此域上的帐户创建",
+  "create_account_button": "创建帐户",
+  "email_input_label": "电子邮件地址",
+  "email_input_placeholder": "你的电子邮件",
+  "name_input_label": "名称",
```
Comment (evroon) [inline frontend/public/locales/zh-CN/common.json:20]: I only know a little Chinese, but I was wondering: what is the difference between this and 名字?

File: frontend/public/locales/en/common.json [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+  "delete_team_button": "Delete Team",
+  "delete_tournament_button": "Delete Tournament",
+  "edit_tournament_button": "Edit Tournament",
+  "recommended_badge_title": "Recommended",
+  "upcoming_matches_empty_table_info": "upcoming matches",
+  "elo_difference": "ELO Difference",
+  "swiss_difference": "Swiss Difference",
+  "could_not_find_any_alert": "Could not find any",
+  "dropzone_accept_text": "Drop files here",
+  "dropzone_reject_text": "PDF file must be less than 10MB",
```
Comment (evroon) [inline frontend/public/locales/en/common.json:210]:
```suggestion
  "dropzone_reject_text": "Image must be less than 10MB",
```
Oh this was already written incorrectly by me before. It's not about PDFs, but images.
could you fix the Chinese translation as well?

**Review by evroon — COMMENTED** [review]
Awesome work!

Translations look good to me overall, I have only small comments.
Could you also cherr-pick these commits onto this branch?
https://github.com/evroon/bracket/commit/c49a99d389b9ff71a91377b27800517050da38a3
https://github.com/evroon/bracket/commit/4201d524278d345a492c43bb349f8a25bd8c7560

The first one fixes `yarn build` which fails now because this `placeholder` parameter isn't allowed anymore apparently. The second just fixes formatting. After that, CI will pass.

File: frontend/public/locales/zh-CN/common.json (hunk 3 above)
Comment (Sevichecc) [reply frontend/public/locales/zh-CN/common.json:20]: I think (名称) is more formal than 名字, and there are no differences in meaning. Maybe 名字 is well suited for this case.

Comment (evroon) [reply frontend/public/locales/zh-CN/common.json:20]: Ah okay thanks

**Comment by Sevichecc** [comment]
Looks like the weclome page is broken, I will fix this tomorrow.

**Comment by evroon** [comment]
@Sevichecc there is this error now:
image

do you want to fix it or shall I finish this PR?

**Comment by Sevichecc** [comment]
@evroon I have committed a fix for that error, maybe I should try to build locally before committing.

File: frontend/src/pages/user.tsx [hunk 5]
```diff
@@ -1,12 +1,21 @@
...
 import UserForm from '../components/forms/user';
 import { checkForAuthError, getTournaments, getUser } from '../services/adapter';
 import { getLogin } from '../services/local_storage';
 import Layout from './_layout';
 
+export const getServerSideProps = async ({ locale }: { locale: string }) => ({
+  props: {
+    ...(await serverSideTranslations(locale, ['common'])),
+  },
+});
```
Comment (evroon) [inline frontend/src/pages/user.tsx:14]: It's a bit annoying maybe that this has to be defined in every page file, but we can look into it later.

**Comment by evroon** [comment]
Thanks a lot again, let's merge this

**Pull request merged**